inst/tests/tests.Rraw

```diff
@@ -1,4 +1,5 @@
 require(methods)
+Sys.setenv("_R_CHECK_LENGTH_1_LOGIC2_" = TRUE)  # Check && and || using only length-1 logicals
```

Comment:
It seems more appropriate to add this in CI instead of test script. Additionally isn't this env var a new thing in R? does it exists on R 3.1.0? Adding it in CI will actually solve issue as we will add it to particular build only.

Reply:
I thought about that, but it won't run for example when `test.data.table()` or revdeps are run. Only for R >= 3.6.0, but setting a variable that R doesn't know about is ok isn't it?

Reply:
By default It won't run unless R will change its default (and it is good, see below), and having that env var fixed in unit test we are dropping tiny part of portability.
Test data.table should be run "as of now R", not a "strict R" or something else. To run strict tests we need to put it in a CI job. If user wants to run strict test, then he set that env var. Having this env var fixed in tests takes this ability away from user.
Setting variable that R don't know and won't use will not make harm, but will make test script misleading. Ideally it could be wrapper in `if (R >= 3.6.0) setenv`, so the code will be clear about its behavior.

Reply:
I don't think any user will want to turn it off.  Setting it here in `tests.Rraw` seems good to me.  It's not misleading especially if there's a comment next to it.

Reply:
Power user could, but lets assume no one will.
`R >= 3.6.0` hasn't been mentioned yet. If we are not doing `if (R >= 3.6.0) setenv` lets at least mention that in comment.